src/platform/amd/renoir/include/platform/fw_scratch_mem.h

```diff
@@ -0,0 +1,121 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+*
+*  Copyright(c) 2021 AMD. All rights reserved.
+*
+*  Author: Basavaraj Hiregoudar <basavaraj.hiregoudar@amd.com>
+*          Anup Kulkarni <anup.kulkarni@amd.com>
+*          Bala Kishore <balakishore.pati@amd.com>
+*/
+#ifndef __EXT_SCRATCH_MEM_H__
+#define __EXT_SCRATCH_MEM_H__
+
+#pragma pack(push, 4)
+
+/* MAX number of DMA descripotos */
+#define	MAX_NUM_DMA_DESC_DSCR	64
+#define SCRATCH_REG_OFFSET	0x1250000
+
+typedef struct _acp_atu_grp_pte_ {
+	uint32_t lowPart;
+	uint32_t highPart;
+} acp_atu_grp_pte_t;
+
+typedef union _acp_cfg_dma_trns_cnt_ {
+	struct{
+		uint32_t  trns_cnt : 19;
+		uint32_t  reserved : 12;
+		uint32_t  ioc : 1;
+	} bits;
+	unsigned int    u32All;
+} acp_cfg_dma_trns_cnt_t;
+
+typedef struct _acp_config_dma_descriptor_ {
+	uint32_t	src_addr;
+	uint32_t	dest_addr;
+	acp_cfg_dma_trns_cnt_t	trns_cnt;
+	uint32_t	reserved;
+} acp_cfg_dma_descriptor_t;
+
+typedef struct _acp_config_dma_misc_ {
+	uint32_t	channelstatus;
+	uint32_t	channel;
+	uint32_t	flag;
+} acp_cfg_dma_misc_t;
+
+typedef struct  _acp_scratch_register_config {
+	/* ACP out box buffer */
+	uint8_t				acp_outbox_buffer[512];
+
+	/* ACP in box buffer */
+	uint8_t				acp_inbox_buffer[512];
+
+	/* ACP debug box buffer */
+	uint8_t				acp_debug_buffer[1024];
+
+	/* ACP exception box buffer */
+	uint8_t				acp_except_buffer[1024];
+
+	/* ACP stream buffer */
+	uint8_t				acp_stream_buffer[1024];
+
+	/* ACP trace buffer */
+	uint8_t				acp_trace_buffer[1024];
+
+	/* Host msg write flag */
+	uint32_t			acp_host_msg_write;
+
+	/* Host ack flag */
+	uint32_t			acp_host_ack_write;
+
+	/* Dsp msg write flag */
+	uint32_t			acp_dsp_msg_write;
+
+	/* Dsp ack flag */
+	uint32_t			acp_dsp_ack_write;
+
+	/* ACP pte1 table */
+	acp_atu_grp_pte_t		acp_atugrp1_pte[16];
+
+	/* ACP pte2 table */
+	acp_atu_grp_pte_t		acp_atugrp2_pte[16];
+
+	/* ACP pte3 table */
+	acp_atu_grp_pte_t		acp_atugrp3_pte[16];
+
+	/* ACP pte4 table */
+	acp_atu_grp_pte_t		acp_atugrp4_pte[16];
+
+	/* ACP pte5 table */
+	acp_atu_grp_pte_t		acp_atugrp5_pte[16];
+
+	/* ACP pte6 table */
+	acp_atu_grp_pte_t		acp_atugrp6_pte[16];
+
+	/* ACP pte7 table */
+	acp_atu_grp_pte_t		acp_atugrp7_pte[16];
+
+	/* ACP pte8 table */
+	acp_atu_grp_pte_t		acp_atugrp8_pte[16];
+
+	/* ACP DMA Descriptor */
+	acp_cfg_dma_descriptor_t	acp_cfg_dma_descriptor[MAX_NUM_DMA_DESC_DSCR];
+
+	/* Stream physical offset  */
+	uint32_t			phy_offset[8];
+
+	/* Stream system memeory size */
+	uint32_t			syst_buff_size[8];
+
+	/* ACP transmit fifo buffer */
+	uint8_t				acp_transmit_fifo_buffer[768];
+
+	/* ACP receive fifo buffer */
+	uint8_t				acp_receive_fifo_buffer[768];
+
+	uint32_t			reserve[];
+} acp_scratch_reg_config_t;
+
+#pragma pack(pop)
```

Comment:
We need to use ```__attribute__``` here - See the IPC headers.

Reply:
Addressed the Review Comment. Done